rfcs/1-declarative-configuration.md

```diff
@@ -0,0 +1,127 @@
+# Declarative Configuration
+
+## Table of Contents
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [Implementation Details](#implementation-details)
+  - [Configuration example](#examples)
+- [Test Plan](#testrelease-plan)
+<!-- /toc -->
+
+
+## Summary
+
+This proposal replaces environment variable based configuration being used in Make+Magefiles in favor of using declarative yaml based configuration.
+
+## Motiviation
+
+As we've been writing `devbase`, and other tooling, we've identified that trying to codify all options for tooling to environment variables, bespoke configuration files, and other obscure means has made it difficult for us to provide options and maintain a common set of toolign.
+
+This, combined with a lack of documentation into all of these options, has laregely made `devbase` unconsumable outside of bootstrap/stencil. Thinking of `devbase` as it's own framework to build software makes it more usable in non-stencil projects (like we do already today w/ submodules)
+### Goals
+
+ - Documentation for all options in configuration file(s)
+ - Standard location for all configuration for tooling provided by devbase, et. al
+ - Documentation mapping 1:1 with Magefile target, with framework to consume/declare it
+ - Documentation on how to write Magefile targets, if the framework docs aren't self-explanatory enough
+ - Support for usage outside of stencil-base, et. al
+ - Not breaking, while we could do a lot more if we did a breaking release, breaking _all_ of the existing releasing tooling isn't optimal
+
+### Non-goals
+
+ - Expose new functionality, this would be nice to have but would balloon the amount of work. We're targeting 1:1 compatibility with features/configuration already exposed today, with nice-to-haves being limited.
```

Comment:
```suggestion
 - Expose new functionality. This would be nice to have, but would balloon the amount of work. We're targeting 1:1 compatibility with features/configuration already exposed today, with nice-to-haves being limited.
```